subprojects/core/src/integTest/groovy/org/gradle/api/tasks/DeferredTaskDefinitionIntegrationTest.groovy

```diff
@@ -311,4 +311,74 @@ class DeferredTaskDefinitionIntegrationTest extends AbstractIntegrationSpec {
         outputContains("Configure :task1")
         failure.assertHasCause("Task with name 'task1' not found")
     }
+
+    def "runs the lazy configuration actions in the same order as the eager configuration actions"() {
+        buildFile << '''
+            def actionExecutionOrderForTaskA = []
+
+            class A extends DefaultTask {}
+
+            tasks.configureEachLater(A) {
+                actionExecutionOrderForTaskA << "A1"
+            }
+
+            tasks.configureEachLater(A) {
+                actionExecutionOrderForTaskA << "A2"
+            }
+
+            def a = tasks.createLater("a", A) {
+                actionExecutionOrderForTaskA << "A3"
+            }
+
+            a.configure {
+                actionExecutionOrderForTaskA << "A4"
+            }
+
+            tasks.configureEachLater(A) {
+                actionExecutionOrderForTaskA << "A5"
+            }
+
+            a.configure {
+                actionExecutionOrderForTaskA << "A6"
+            }
+
+            def actionExecutionOrderForTaskB = []
+
+            class B extends DefaultTask {}
+
+            tasks.withType(B) {
+                actionExecutionOrderForTaskB << "B1"
+            }
+
+            tasks.withType(B) {
+                actionExecutionOrderForTaskB << "B2"
+            }
+
+            def b = tasks.create("b", B) {
+                actionExecutionOrderForTaskB << "B3"
+            }
+
+            b.configure {
+                actionExecutionOrderForTaskB << "B4"
+            }
+
+            tasks.withType(B) {
+                actionExecutionOrderForTaskB << "B5"
+            }
+
+            b.configure {
+                actionExecutionOrderForTaskB << "B6"
+            }
+
+            task assertActionExecutionOrder {
+                dependsOn a, b
+                doLast {
+                    assert actionExecutionOrderForTaskA == actionExecutionOrderForTaskB
```

Comment:
How are these equal?  Wouldn't one contain `[ A1, A2, ... ]` and the other `[ B1, B2, ... ]`?

We should also assert that the right number of events have been received.

Reply:
I'm so confused. I swear I ran the test and it was green... That doesn't remove the fact that it was pretty dumb mistake on my part.